### PR TITLE
First draft at the README

### PR DESCRIPTION
**Pull request by JimPatterson**
First draft, more to come...

### CODE REVIEW

**Comment by JimPatterson** [comment]
@jpamp, here is a first draft of a readme and other project documentation.  I'll be updating this as I go.  Feedback would be appreciated.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,89 @@
...
+your company and/or your company's customers.
+
+We at De Re CRUD think that the mechanical operation of providing CRUD
+APIs and screens with proper validation, error handling, security, and
+lots more can be written once and simply applied to your data.  As a
+side benefit, you will get tools to help you define, document, and
+publish the schema for your model.  And for just three additional
+payments, no wait, I've seen too many infomercials, this is open source,
+you don't have to pay to use this.  Seriously though, you can use the
+APIs internally and get same features in the parts you build.
```
Comment (jpamp) [inline README.md]: and get *the* same features

File: README.md [hunk 2]
```diff
@@ -1,2 +1,89 @@
...
+model.  We include much more information in the schema than most systems
+you have likely used, but it is not more information than you have, you
+just did not put it all in one place.  In addition to allowing us to
+generate APIs, screens, documentation, and more, putting all this information
+together is just a good idea.  Your schema will benefit from
+[SPoT/SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth),
+[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), and
+[DiE](https://www.rallydev.com/blog/engineering/clean-code-duplication-evil)
+principles.  Aside from the obvious benefits of consistent behaviours
+and appearance (e.g. changing every occurrence of field's prompt by
```
Comment (jpamp) [inline README.md]: occurrence of *a* field's prompt

File: README.md [hunk 3]
```diff
@@ -1,2 +1,89 @@
...
+In most cases, your schema will be a logical view of the information,
+but the real world requirements of storing that data will impose
+restrictions on your layout.  De Re CRUD will provide a simple mapping
+between your logical schema and your physical storage.
+
+### Step #3 - Configure the server
+
+We will be providing a server (likely [Node.js](https://nodejs.org/)
+based) that will serve the APIs and UIs.  We plan on a modular approach,
+so alternate server implementations are possible.
```
Comment (jpamp) [inline README.md]: How about adding, "Don't see a server implementation in your preferred language?  Go forth an write one using our documentation as a reference."  - or something like that.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,89 @@
...
+The server will require access to your data storage, so you will need to
+configure the storage access via pluggable storage engines.  We will be
+providing some standard storage engines, and the interface will be well
+documented allowing additional engines to be written.
+
+### Step #4 - Choose your services
+
+The pluggable nature of De Re CRUD will allow you to choose which
+services you would like to use.  Each major feature will be a separate
+module, so if you don't want APIs, simple disable that module.  Want to
```
Comment (jpamp) [inline README.md]: want APIs, *simply* disable that module.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,89 @@
...
-The model documentation
+# De Re CRUD
+#### Never write a CRUD API or maintenance screen again!
+
+If you are a developer building business apps, the odds are that you
+and/or your colleagues spend too much time building and maintaining
+basic
+[CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete)
+APIs and maintenance screens for all the data your app needs.  The worst
+part is that a good part of it is not even used in your app.  A
```
Comment (KatFrog) [inline README.md]: This sentence is a bit confusing.  Maybe use, "The worst part is that a significant percentage of data is not put to use in your app, merely stored and displayed on command (e.g. description fields, notes, instructions, comments, ...)."

File: README.md [hunk 6]
```diff
@@ -1,2 +1,89 @@
...
+significant portion of the data in most business apps is just stored
+and displayed with any real value added by the software (think of all
+the description fields, notes, instructions, comments, etc).  Your time
+is usually the most expensive resource for software companies.  That
+time should be spent writing software that adds significant value to
+your company and/or your company's customers.
+
```
Comment (KatFrog) [inline README.md]: Add a comment here that DeRe CRUD allows you to separate/isolate your business logic from the data.

**Comment by JimPatterson** [comment]
Okay, I made the suggested changes.

**Comment by KatFrog** [comment]
I like it.

**Pull request merged**